pkg/operator2/configsync.go

```diff
@@ -91,64 +91,189 @@ type sourceData struct {
 	mount  corev1.VolumeMount
 }
 
+// TODO: new source data could be generalized
+func newSourceDataIDPSecret(index int, secretName, idpType string) (string, sourceData) {
+	dest := getIDPName(index, secretName, idpType)
+	var ret sourceData
+
+	volume, mount, path := secretVolume(index, dest, idpType)
+	ret.src = secretName
+	ret.path = path
+	ret.volume = volume
+	ret.mount = mount
+
+	return dest, ret
+}
+
+func newSourceDataIDPConfigMap(index int, cmName, idpType string) (string, sourceData) {
+	dest := getIDPName(index, cmName, idpType)
+	var ret sourceData
+
+	volume, mount, path := secretVolume(index, dest, idpType)
+	ret.src = cmName
+	ret.path = path
+	ret.volume = volume
+	ret.mount = mount
```

Comment:
Same nit about removing var and mutations in favor of setting in new struct directly